cockatrice/src/userconnection_information.cpp

```diff
@@ -0,0 +1,72 @@
+#include "userconnection_information.h"
+
+UserConnection_Information::UserConnection_Information()
+{
+}
+
+void UserConnection_Information::addNewServer(QString _saveName, QString _serv, QString _port, QString _username, QString _password, bool _savePassword)
+{
+    QSettings settings("Cockatrice", "server_details");
+
+    settings.beginGroup("server_details");
+    settings.beginWriteArray("server_details");
+    settings.setValue("saveName", saveName);
+    settings.setValue("server", server);
+    settings.setValue("port", port);
+    settings.setValue("username", username);
+    settings.setValue("password", password);
+    settings.setValue("savePassword", savePassword);
+    settings.endGroup();
+}
+
+QMap<QString, UserConnection_Information> UserConnection_Information::getServerInfo()
+{
+    QMap<QString, UserConnection_Information> serverList = new QMap<QString, UserConnection_Information>();
+    
+    int size = settings.beginReadArray("server_details");
+    for (int i = 0; i < size; i++)
+    {
+        settings.setArrayIndex(i);
+        
+        QString saveName = settings.value("saveName").toString();
+        QString serverName = settings.value("server").toString();
+        QString portNum = settings.value("port").toString();
+        QString userName = settings.value("username").toString();
+        QString pass = settings.value("password").toString();
+        bool savePass = settings.value("savePassword").toBool();
+        
+        UserConnection_Information userInfo(saveName, serverName, portNum, userName, pass, savePass);
+        
+        serverList.insert(saveName, userInfo);  
+    }
+
+    return serverList;
+}
+
+QStringList UserConnection_Information::getServerInfo(const QString &find)
+{
+    QStringList server = new QMap<QString, UserConnection_Information>();
```

Comment:
The declared types don't match: a `QStringList` is a `QList<QString>`, not a `<QMap>`;
Also, when you call "new" you are allocating a new object to memory, so the result is a *pointer